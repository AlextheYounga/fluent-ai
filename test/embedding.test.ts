import { test, expect } from "bun:test";
import { openai, ollama, voyageai, requestObject, load } from "../src";

const jobs = [
  ollama().embedding("nomic-embed-text"),
  openai({ apiKey: "<key>" }).embedding("text-embedding-ada-002"),
  voyageai({ apiKey: "<key>" }).embedding("voyage-3-lite"),
];

test("embeddings", async () => {
  for (const job of jobs) {
    expect(
      await requestObject(job.input("hi").makeRequest())
    ).toMatchSnapshot();
  }
});

test("dump", () => {
  for (const job of jobs) {
    expect(job.dump()).toMatchSnapshot();
  }
});

test("load", async () => {
  for (const job of jobs) {
    const req1 = await requestObject(load(job.dump()).makeRequest!());
    const req2 = await requestObject(job.makeRequest());
    expect(req1).toEqual(req2);
  }
});
