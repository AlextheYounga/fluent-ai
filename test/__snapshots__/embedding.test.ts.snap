// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`embeddings 1`] = `
{
  "body": "{"model":"nomic-embed-text","input":"hi"}",
  "headers": {},
  "method": "POST",
  "url": "http://localhost:11434/api/embed",
}
`;

exports[`embeddings 2`] = `
{
  "body": "{"model":"text-embedding-ada-002","input":"hi"}",
  "headers": {
    "authorization": "Bearer <key>",
    "content-type": "application/json",
  },
  "method": "POST",
  "url": "https://api.openai.com/v1/embeddings",
}
`;

exports[`embeddings 3`] = `
{
  "body": "{"model":"voyage-3-lite","input":"hi"}",
  "headers": {
    "authorization": "Bearer <key>",
    "content-type": "application/json",
  },
  "method": "POST",
  "url": "https://api.voyageai.com/v1/embeddings",
}
`;

exports[`dump 1`] = `
{
  "embedding": {
    "model": "nomic-embed-text",
    "params": {
      "input": "hi",
    },
  },
  "options": {},
  "provider": "ollama",
}
`;

exports[`dump 2`] = `
{
  "embedding": {
    "model": "text-embedding-ada-002",
    "params": {
      "input": "hi",
    },
  },
  "options": {
    "apiKey": "<key>",
  },
  "provider": "openai",
}
`;

exports[`dump 3`] = `
{
  "embedding": {
    "model": "voyage-3-lite",
    "params": {
      "input": "hi",
    },
  },
  "options": {
    "apiKey": "<key>",
  },
  "provider": "voyageai",
}
`;
